torchrl/envs/transforms/transforms.py

```diff
@@ -4589,8 +4619,25 @@ def transform_input_spec(self, input_spec: TensorSpec) -> TensorSpec:
     def _batch_size(self):
         return self.parent.batch_size
 
+    def _validate_value_tensor(self, value, spec):
+        if value.shape != spec.shape:
+            raise RuntimeError(
+                f"Value shape ({value.shape}) does not match the spec shape ({spec.shape})."
+            )
+        if value.dtype != spec.dtype:
+            raise RuntimeError(
+                f"Value dtype ({value.dtype}) does not match the spec dtype ({spec.dtype})."
+            )
+        if value.device != spec.device:
+            raise RuntimeError(
+                f"Value device ({value.device}) does not match the spec device ({spec.device})."
+            )
+        if not spec.is_in(value):
```

Comment:
Here it is, I don't think this can be false after the previous checks but it does the same job